vllm/core/scheduler.py

```diff
@@ -1445,7 +1481,18 @@ def _get_num_lookahead_slots(self, is_prefill: bool) -> int:
         lookahead allocation for prefill.
         """
         if is_prefill:
-            return 0
+            if self.scheduler_config.is_multi_step_chunked_prefill:
+                # num_lookahead_slots was introduced in the context of decodes,
+                # in Speculative Decoding.
+                # When the num_scheduler_steps is 8, say, then the
+                # num_lookahead_slots is 7. Meaning, we are doing a 1-step of
+                # decode anyways and we wish to do 7 more.
+                #
+                # "lookaheads" for prefills, is introduced in support for
+                # Chunked-Prefill in Multi-Step.
```

Comment:
No need to provide rationale of lookahead slots. Instead, just comment "why prefill needs lookahead slots" here

Reply:
I added the explanation for the decode case so the the + 1 in `self.scheduler_config.num_lookahead_slots + 1` makes sense. does it do more harm than good, do you think ?